x-pack/plugins/cases/public/components/markdown_editor/plugins/lens/use_lens_open_visualization.tsx

```diff
@@ -38,15 +41,26 @@ export const useLensOpenVisualization = ({ comment }: { comment: string }) => {
     );
   }, [lensVisualization, navigateToPrefilledEditor]);
 
+  if (!lensVisualization.length || lensVisualization?.[0]?.attributes == null) {
+    return { canUseEditor: hasLensPermissions, actionConfig: null };
+  }
+
+  const lensAttributes = lensVisualization[0]
+    .attributes as unknown as TypedLensByValueInput['attributes'];
+
+  const isESQLQuery = isOfAggregateQueryType(lensAttributes.state.query);
```

Comment:
I checked this function's implementation and it looks for `'sql' in query || 'esql' in query`.

I noticed in [the test](https://github.com/elastic/kibana/pull/171191/files#diff-3d855e8724e990081b24a1c961a69656c9d9b56a51e48d1b9ceddb5e7137c463R71) you only set `esql`.

Now, I have no idea so I am asking 😅  do both of these qualify as `ES|QL`? Could `isESQLQuery` be `true` when it shouldn't?

Reply:
Before adding es|ql we were supporting sql but we removed it when es|ql came! So this function checks for both (sql and esql) but only esql queries are used in the charts and everywhere else. I want to clean it up but I can assure you that is not used anywhere :)

Reply:
All good then, thank you! 🙌